GitUI/CommandsDialogs/SettingsDialog/ShellExtension/ShellExtensionManager.cs

```diff
@@ -0,0 +1,153 @@
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Diagnostics;
+using System.IO;
+using System.Reflection;
+using GitCommands;
+using Microsoft.Win32;
+
+namespace GitUI.CommandsDialogs.SettingsDialog.ShellExtension
+{
+    public static class ShellExtensionManager
+    {
+        /// <summary>
+        /// Checks if necessary for registration files were found
+        /// </summary>
+        /// <returns>false only if all necessary files were found, otherwise true</returns>
+        public static bool IsNotInstalled()
+        {
```

Comment:
```suggestion
        /// <summary>
        ///  Checks if the shell extension files are found at the known location.
        /// </summary>
        /// <returns><see langword="true"/> the extension files are found at the known location; otherwise <see langword="false"/>.</returns>
        public static bool CheckFilesFound()
        {
```